internal/broadcaster/utxo_creator.go

```diff
@@ -7,153 +7,149 @@ import (
 	"errors"
 	"fmt"
 	"log/slog"
+	"sync"
 	"time"
 
-	sdkTx "github.com/bitcoin-sv/go-sdk/transaction"
-
 	"github.com/bitcoin-sv/arc/internal/metamorph/metamorph_api"
 	"github.com/bitcoin-sv/arc/pkg/keyset"
+	sdkTx "github.com/bitcoin-sv/go-sdk/transaction"
 )
 
 type UTXOCreator struct {
 	Broadcaster
-	keySets []*keyset.KeySet
+	keySet *keyset.KeySet
+	wg     sync.WaitGroup
 }
 
-func NewUTXOCreator(logger *slog.Logger, client ArcClient, keySets []*keyset.KeySet, utxoClient UtxoClient, isTestnet bool, opts ...func(p *Broadcaster)) (*UTXOCreator, error) {
+func (b *UTXOCreator) Shutdown() {
+	b.cancelAll()
+	b.wg.Wait()
+}
```

Comment:
For consistency with other consolidator and rate broadcaster move this function to the bottom of this file